### PR TITLE
Add Monitor for Operator + Apply Standards

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Motivation
- Apply the same standard used in the other projects as UPS Oper in order to add the Monitoring features easily. (https://issues.jboss.org/browse/AEROGEAR-9598)
- https://issues.jboss.org/browse/AEROGEAR-9190
- https://issues.jboss.org/browse/AEROGEAR-9189

## What
- Add default prometheus and graphana dashboard for the operator. 
- Add GitHub files from standards ( same applied in all projects ) 
- Same Makefile structure of UPS Oper
- Same README structure of UPS Oper
- Add unit test, coveralls and CI as UPS Oper
- Add release process as UPS Oper
- Add e2e test as it was done for UPS Oper
- Improve APP NAMESPACE and OPER namespace validation in order to allow unit test and run it locally + unit tests to ensure that it is working as it was done for other projects. 

## Why
- Be able to easily create the monitor command and apply changes horizontally
- Follow the standards and make easier to keep maintained
- Add requirements to create the SOP's and Monitor features  

## Steps to verify
- **Check the README changes;** ( same ups oper structure )
- **Check that the CI working:**
- **Check that the Jenkins is working here with the e2e test as well now**.
- Check the Graphana dash here: https://grafana-route-middleware-monitoring.apps.london-2f09.openshiftworkshop.com/d/null/app-metrics-operator?refresh=10s&orgId=1
- Check the Prometheus Rules/Alerts here: https://prometheus-route-middleware-monitoring.apps.london-2f09.openshiftworkshop.com/alerts

Screenshot 2019-07-16 at 07 03 23

Screenshot 2019-07-16 at 07 02 20

Screenshot 2019-07-15 at 16 25 52

## Checklist:
- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Progress
- [x] Finished task
- [ ] TODO

### CODE REVIEW

**Comment by grdryn** [comment]
@camilamacedo86 could you indicate what parts of #11 block this? There seem to be lots of different things in that one, and I don't really have time to look at all of those at the moment. This PR is a higher priority than #11, so I just want to make sure I understand what the dependency is.

**Comment by camilamacedo86** [comment]
Hi @grdryn,
It is regarding the Makefile commands and in the README organization/info as well. 

Also, note that we should be able to merge the https://github.com/aerogear/app-metrics-operator/pull/11 easily it just applying the standard and keep it as  == UPS Oper structure as well. It is not really changing how things work.

**Comment by camilamacedo86** [comment]
Hi @grdryn, @aliok, @wei-lee 

I know that it is the backlog. However, would be great with we could just merge it for do not lose the work performed here and then we will be able to do any further required change after when it get out from our backlog.  Could we move forward in this way? IMHO, it just brings good things for the project and it is in the init dev process so should not be a big problem at all. 

Really tks for your attention and all support.

**Comment by camilamacedo86** [comment]
Closing since shows outdated and could not be verified/merged so far.

**Comment by camilamacedo86** [comment]
Hi @grdryn I centralized all in the same PR and rebased it with the master in order to solve the CI issues.

File: Gopkg.lock [hunk 1]
```diff
@@ -56,6 +56,25 @@
   pruneopts = "NUT"
   revision = "de5bf2ad457846296e2031421a34e2568e304e35"
 
+[[projects]]
```
Comment (grdryn) [inline Gopkg.lock]: These two entries look wrong -- I don't believe there should be any imports from the mobile-security-service-operator or unifiedpush-operator in here (and I can't see anywhere in this PR where they're used).

**Review by grdryn — CHANGES_REQUESTED** [review]
Looks fine apart from the stray dependencies in the Gopkg.lock.

File: Gopkg.lock (hunk 1 above)
Comment (camilamacedo86) [reply Gopkg.lock]: Fixed with dep manager. Probably it missed up with the rebases.

**Review by grdryn — APPROVED** [review]
:+1:

**Pull request merged**